### ISSUE TITLE
Track making: Normal Maps not displayed correctly

### ISSUE THREAD

**Issue by JulianSchoenbaechler** [supertuxkart/stk-code#2243]
I'm trying to use normal maps in my track but they don't work properly (See attached image). I'm adding normal maps in the image properties of the texture and they're working so far.
What I've done:
- Making the mesh where the normal map should apply a seperate object.
- Tried with a second UV-Layer with the normal map.
- Changing the size of the textures.
- Checked normals and mesh.
- Changed sun position (which doesn't affect the position of this strange line on the picture).

As I don't see any mistake on my side so far -> I suppose it's just a bug.

![normalmap_issue](https://cloud.githubusercontent.com/assets/10711264/8751744/5509b416-2caf-11e5-8271-01074ea50c91.jpg)

**Comment by CreativeForge** [comment]
Little Update:
- Triangulated the mesh:

![normalmap_issue2](https://cloud.githubusercontent.com/assets/10711264/8754097/5dc8d6f6-2cc1-11e5-941f-8382e8a82af2.jpg)

**Comment by auriamg** [comment]
Could you please share the track and the blend so that we can take a look?

**Comment by CreativeForge** [comment]
I'm sorry, I wasn't able to write back until now...
I found the reason for this shading issue. It has to do with the UV mapping (thanks for the tip to check them, samuncle). Irrlicht (and also other graphic engines, even the blender game engine sometimes) seems to have problems when the UV faces overlaps each other.
In my case my UV faces overlaps multiple times -> For example, I used to create the railroads a mirror modifier and an array modifier (+ curve modifier to follow the path). The UV map is therefore copied mutliple times. To avoid this problem I tried to increment the uv cordinates for every copy, so that the uv faces won't overlap but keeping the same texture as before. I also experimented with the UV Project modifier but without very good results.
Anyway it would be a huge amount of extra work for me to get normal maps working on objects with overlapping UV faces. So I think I won't go into this issue any further and just use normal mapping on smaller track objects.
